publish-sdk

```diff
@@ -51,64 +59,70 @@ update_image() {
   local remote_alt_arch_image="${remote_image}-${SHORT_SHA}-${alt_arch}"
 
-  if ! docker image inspect "${local_host_arch_image}" >/dev/null 2>&1 ; then
-    echo "did not find ${local_host_arch_image}, skipping it ..." >&2
-    return
-  fi
-
-  if ! docker tag "${local_host_arch_image}" "${remote_host_arch_image}" ; then
-    echo "failed to tag ${remote_host_arch_image}" >&2 && exit 1
+  if [[ "${ONLY_MANIFEST:-"false"}" != "true" ]]; then
+    if ! docker image inspect "${local_host_arch_image}" >/dev/null 2>&1 ; then
+      echo "did not find ${local_host_arch_image}, skipping it ..." >&2
+      return
+    fi
+
+    if ! docker tag "${local_host_arch_image}" "${remote_host_arch_image}" ; then
+      echo "failed to tag ${remote_host_arch_image}" >&2 && exit 1
+    fi
+
+    # push the image, if it exists locally
+    if ! docker push "${remote_host_arch_image}" ; then
+      echo "failed to push ${remote_host_arch_image}" >&2 && exit 1
+    fi
+    ((++ECR_OPERATIONS))
   fi
 
-  # push the image, if it exists locally
-  if ! docker push "${remote_host_arch_image}" ; then
-    echo "failed to push ${remote_host_arch_image}" >&2 && exit 1
-  fi
-
-  # clean up any cached local copy of the manifest
-  manifest_dir="${HOME}/.docker/manifests"
-  mkdir -p "${manifest_dir}"
-  find "${manifest_dir}" -type f -delete
-  find "${manifest_dir}" -mindepth 1 -type d -delete
-
-  if ! docker manifest inspect "${remote_alt_arch_image}" >/dev/null 2>&1 ; then
-    echo "could not find ${remote_alt_arch_image}, skipping manifest creation" >&2
-    return
-  fi
-
-  local image="${remote_image##*/}"
-  local repo="${image%:*}"
-
-  # clean up any remote (ECR) copy of the manifest
-  if [[ "${REGISTRY}" =~ ^public\.ecr\.aws\/ ]]; then
-    aws ecr-public batch-delete-image \
-      --repository-name "${repo}" \
-      --image-ids imageTag="${VERSION}" \
-      --region "us-east-1"
-  elif [[ "${REGISTRY}" =~ ^[0-9]+\.dkr\.ecr\.[a-z0-9-]+\.amazonaws\.com$ ]]; then
-    aws ecr batch-delete-image \
-      --repository-name "${repo}" \
-      --image-ids imageTag="${VERSION}"
-  fi
-
-  if ! docker manifest create "${remote_image}" \
-      "${remote_host_arch_image}" "${remote_alt_arch_image}" ; then
-    echo "failed to create manifest ${remote_image}" >&2 && exit 1
-  fi
-
-  if ! docker manifest annotate "${remote_image}" \
-      "${remote_host_arch_image}" --arch "${docker_arch}" ; then
-    echo "failed to annotate manifest ${remote_image} with ${remote_host_arch_image}" >&2 && exit 1
-  fi
-
-  if ! docker manifest annotate "${remote_image}" \
-      "${remote_alt_arch_image}" --arch "${alt_docker_arch}" ; then
-    echo "failed to annotate manifest ${remote_image} with ${remote_alt_arch_image}" >&2 && exit 1
-  fi
-
-  # push the manifest and remove the local cache
-  if ! docker manifest push --purge "${remote_image}" ; then
-    echo "failed to push ${remote_image}" >&2 && exit 1
+  if [[ "${SKIP_MANIFEST:-"false"}" != "true" ]]; then
```

Comment:
You can invert the conditional and just return early if `SKIP_MANIFEST` is "true". That will avoid a lot of the indentation-related changes.